core/src/modules/cloud.ts

```diff
@@ -0,0 +1,149 @@
+import os from "os";
+import path from "path";
+import tar from "tar";
+import fs from "fs";
+import FormData from "form-data";
+import { mkdtemp, copy, remove } from "fs-extra";
+import axios, { AxiosError, AxiosInstance } from "axios";
+
+export class CloudError extends Error {
+  code: string;
+
+  constructor({ code, message }: { code: string; message: string }) {
+    super(message);
+    this.code = code;
+  }
+}
+
+export async function packageConfig(
+  projectPath: string,
+  configDirPath: string,
+  tarballPath: string
+) {
+  const tempDir = await mkdtemp(path.join(os.tmpdir(), "grouparoo-pack-"));
```

Comment:
trailing `-`?  Also, should we use a new tmpdir for each pack attempt?

Reply:
We are! `mkdtemp` actually makes a new directory. `grouparoo-pack-` is a prefix, to which `mkdtemp` appends a random series of characters. Here's an example one it generated: 
`/private/var/folders/1m/m2fpq2hx6t90xbh_j52j8cq00000gn/T/grouparoo-pack-OmQnwR`